src/structures/Guild.js

```diff
@@ -1422,6 +1485,54 @@ class Guild extends Base {
       .then(() => this);
   }
 
+  /**
+   * Edits the guild's membership screening form.
+   * @param {GuildMembershipScreeningData} memberScreen The membership screening data for the guild
+   * @returns {Promise<GuildMembershipScreening>}
+   */
+  setMembershipScreening(memberScreen) {
+    if (!this.features.includes('COMMUNITY')) {
+      throw new Error('COMMUNITY');
+    }
+    let form_fields = [];
+    if (typeof memberScreen.formFields !== 'undefined') {
+      for (const field of memberScreen.formFields) {
+        form_fields.push({
+          field_type: field.field_type,
+          label: field.label,
+          values: field.values,
+          required: field.required,
+        });
+      }
+    }
+    let data = {};
+    if (typeof memberScreen.enabled !== 'undefined') data.enabled = memberScreen.enabled;
+    if (typeof memberScreen.description !== 'undefined') data.description = memberScreen.description;
+    if (typeof memberScreen.formFields !== 'undefined') data.form_fields = JSON.stringify(form_fields);
+    return this.client.api
+      .guilds(this.id, 'member-verification')
+      .patch({
+        data: data,
+      })
+      .then(res => {
+        this.membershipScreeningEnabled = memberScreen.enabled;
+        let formFields = [];
+        for (const field of res.form_fields) {
+          formFields.push({
+            fieldType: field.field_type,
+            label: field.label,
+            values: field.values,
+            required: field.required,
+          });
+        }
+        return {
+          enabled: this.membershipScreeningEnabled,
+          description: res.description,
+          formFields: formFields,
```

Comment:
seems this code is duplicated, in any case the same as before this can use the `map` function

Reply:
I noticed that there was a few instances of this necessary, would it be desirable to have a private function that transforms this data and call that in each location instead?